.github/workflows/build.yml

```diff
@@ -1,23 +1,57 @@
-name: Frontend
+name: SonarCloud Analysis
 
 on:
   push:
     branches:
+      - 40-sonarcloud-integration
       - main
       - dev
   pull_request:
     types: [opened, synchronize, reopened]
 
 jobs:
-  sonarcloud:
+  backend:
+    name: Backend
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - name: Checkout code
+        uses: actions/checkout@v4
         with:
-          fetch-depth: 0  
+          fetch-depth: 0
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: 17
+          distribution: 'zulu'
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
```

Comment:
Remove this and try to run it.
If it doesn't work then put it back.

Reply:
It works without it. That piece of code was included because when I created the project on SonarCloud, they provided a base build.yml file that already had it included. At the time, I thought it was important.